### ISSUE TITLE
[question] Is it possible or what's your advise to check if a file is already minified?

### ISSUE THREAD

**Issue by mikegleasonjr** [tdewolff/minify#110]
Either already minified with your minifier or any 3rd party ones?

**Comment by tdewolff** [comment]
No I don't think so. What you could do is check if the filesize changes when you minify. However, other minifiers often don't minify as thoroughly as these (not for JS though), so these minifiers will still scrape off some bytes.

I would put minified files in a different directory, namely the production-ready folder instead of the development folders. Or you could append `-min` to the filename or something similar.

So in short, no you don't know beforehand if it is minified unless you use a file naming convention or a different location.

**Comment by tomByrer** [comment]
(I had volunteered with [jsDelivr CDN](http://www.jsdelivr.com/) )

Typical convention is a minified file will have `.min` before the file extension, eg `my-js-file.min.js`  I *personally prefer* the original concocted file to be `my-js-file.src.js` & the minified file `my-js-file.js`; but you'll see that in only 2% of the repos.

It is common to have a `/src` (source) & a `/dist` (distribution use) folder for web-facing & node-deployed repos.

I had thought about making a minified tester by checking the amount of whitespace after the [first-line comment banner](https://github.com/tomByrer/web-engeneering-notes/blob/master/banner-tips.md).  Would that be of interest @mikegleasonjr

**Comment by mikegleasonjr** [comment]
Thanks guys, I also read some techniques like checking the line length of the first few lines. Usually they'll be pretty long.

The file naming convention (.min) is bar far the lowest hanging fruit (and pretty juicy too 😄 ) because in my tests cases, it would cover at least 50% of my files. Saving 50% minification is a huge win.

@tomByrer I would be interested for sure but like I said for now the .min is good enough! Thanks for helping though!

@tdewolff By the way, can't wait to deliver my project so I can show it to you... You are of a tremendous help with your library and quick bug fixes!
